File: Homework_OOP_2/Dog.java
package Homework_OOP_2;

public class Dog extends HomeAnimal {
    private final String training;

    protected Dog(String nickname, String breed, String vaccination, String coatColor, String dateBirthday, int height,
            int weight, String colorEye, String training) {
        super(nickname, breed, vaccination, coatColor, dateBirthday, height, weight, colorEye);
        this.training = training;
    }

    public String getTraining() {
        return this.training;
    }

    public void trainDog() {
        System.out.println("Идет дрессировка");
    }

    @Override
    public void showAffection() {
        System.out.println("Мне нужна твоя ласка");
    }

    @Override
    public void animalSound() {
        System.out.println("Гавввввв");
    }

    @Override
    public String toString() {
        return String.format("Собака: %s; Дрессировка: %s", super.toString(), getTraining());
    }
}

File: Homework_OOP_7/classes/home/dog.py
from ..interfeces.dogTraining import DodTraining
from .home import Home

class Dog(Home, DodTraining):
    def __init__(self, name, height, weight, colorEye, name2, breed, vaccination, color, date, wool):
        self.wool = wool
        super(Dog, self).__init__(name, height, weight, colorEye, name2, breed, vaccination, color, date)

    def animalSay(self):
        print(f"{super().animalSay()} 'Гав-гав'")

    def caressing(self):
        return(super().caressing())
    
    def training(self):
        DodTraining.training(self)

    def __str__(self) -> str:
        return (f"{super().__str__()} Наличие шерсти: {self.wool};")